designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ScenarioActivityManager.scala

```diff
@@ -0,0 +1,41 @@
+package pl.touk.nussknacker.engine.api.deployment
+
+import pl.touk.nussknacker.engine.api.deployment.ScenarioActivityManager.ScenarioActivityModificationResult
+
+import scala.concurrent.Future
+
+trait ScenarioActivityManager {
+
+  def saveActivity(
+      scenarioActivity: ScenarioActivity
+  ): Future[Unit]
+
+  def modifyActivity(
+      scenarioActivityId: ScenarioActivityId,
+      modify: ScenarioActivity => ScenarioActivity,
+  ): Future[ScenarioActivityModificationResult]
+
+}
+
+object ScenarioActivityManager {
+  sealed trait ScenarioActivityModificationResult
```

Comment:
`ModificationResult` is enough ;) (the know that the context is scenario activity due to the fact the ADT is placed in the ScenarioActivityManager companion object)